addons/l10n_it_edi/data/invoice_it_template.xml

```diff
@@ -204,11 +204,11 @@
 
 <template id="account_invoice_it_FatturaPA_sede">
             <Sede>
-                <Indirizzo><t t-if="partner.street" t-esc="format_alphanumeric(partner.street)"/> <t t-if="partner.street2" t-esc="format_alphanumeric(partner.street2)"/></Indirizzo>
-                <CAP><t t-if="partner.country_id.code != 'IT'" t-esc="'00000'"/><t t-else="" t-esc="partner.zip"/></CAP>
-                <Comune t-esc="format_alphanumeric(partner.city)"/>
-                <Provincia t-if="partner.country_id.code == 'IT'" t-esc="partner.state_id.code"/>
-                <Nazione t-esc="partner.country_id.code"/>
+                <Indirizzo><t t-if="partner.street or partner.street2" t-esc="format_alphanumeric((partner.street or '') + ' ' + (partner.street2 or ''))[:60]"/></Indirizzo>
+                <CAP><t t-if="partner.country_id.code != 'IT'" t-esc="'00000'"/><t t-elif="partner.zip" t-esc="partner.zip[:5]"/></CAP>
+                <Comune t-esc="format_alphanumeric(partner.city[:60])"/>
+                <Provincia t-if="partner.country_id.code == 'IT' and partner.state_id" t-esc="partner.state_id.code[:2]"/>
```

Comment:
I wonder if the ands are needed .  It might check already if state_id exists

Reply:
If the country is not 'IT', then the state must be ignored and the tag shouldn't be there.
Suppose it's an US partner, the State will be present (i.e. Washington and it must be ignored.)

From the code in `res.country.state` it doesn't look like there's a limit on the code length, but Italian "states" (province) all have 2 chars, and they are inserted in `res.country.state.csv` in the base module, so... it should already be correct anyway.